modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php

```diff
@@ -208,8 +210,17 @@ public function updateEmbedTag($id, $type, $tag, $view, \DOMDocument $dom, \DOME
    */
   public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
 
-    $dom = new \DOMDocument();
-    $dom->loadHTML($value, LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD);
+    $value_is_array = is_array($value);
+    $text = (string) ($value_is_array ? $value['value'] : $value);
+    if (strpos($text, '<drupal-entity ') === FALSE) {
+      return $value;
```

Comment:
Shouldn't this be returning `$text` instead of `$value`?
```suggestion
      return $text;
```